src/ml/neural_net/image_augmentation.cpp

```diff
@@ -14,6 +14,42 @@
 namespace turi {
 namespace neural_net {
 
+shared_float_array convert_to_shared_float_array(
+    std::vector<image_annotation> annotations_per_image) {
+  size_t offset = 0;
+  std::vector<float> ann(annotations_per_image.size() * 6);
+  for (size_t j = 0; j < annotations_per_image.size(); j++) {
+    offset = j * 6;
+    image_annotation annotation = annotations_per_image[j];
+    ann[offset] = annotation.identifier;
+    ann[offset + 1] = annotation.bounding_box.x;
+    ann[offset + 2] = annotation.bounding_box.y;
+    ann[offset + 3] = annotation.bounding_box.height;
+    ann[offset + 4] = annotation.bounding_box.width;
+    ann[offset + 5] = annotation.confidence;
+  }
+  shared_float_array data_to_augment =
+      shared_float_array::wrap(std::move(ann), {annotations_per_image.size() * 6});
+  return data_to_augment;
+}
+
+std::vector<image_annotation> convert_to_image_annotation(
+    shared_float_array augmented_annotation) {
+  const size_t* shape = augmented_annotation.shape();
+  std::vector<image_annotation> augmented_ann;
```

Comment:
You haven't actually allocated the vector here. I think you want
```
std::vector<image_annotation> augmented_ann(shape[0]);
```
Although it would be nice to factor our `shape[0]` and give it a more readable name